objects/src/transaction/inputs.rs

```diff
@@ -327,64 +326,95 @@ fn build_nullifier_commitment<T: ToNullifier>(notes: &[T]) -> Digest {
 // INPUT NOTE
 // ================================================================================================
 
+const AUTHENTICATED: u8 = 0;
+const UNVERIFIED: u8 = 1;
+
 /// An input note for a transaction.
 #[derive(Debug, Clone, PartialEq, Eq)]
 #[cfg_attr(feature = "serde", derive(serde::Deserialize, serde::Serialize))]
-pub struct InputNote {
-    note: Note,
-    proof: NoteInclusionProof,
+pub enum InputNote {
+    /// Input notes which have an authentication path and are authenticated by the transaction
+    /// kernel.
+    Authenticated { note: Note, proof: NoteInclusionProof },
+
+    /// Input notes which do not yet an authentication path. These are consumed by the transaction
+    /// kernel, and later on authenticated by the rollup kernels.
```

Comment:
nit: there is a "have" missing after 'yet". But also, maybe this could be: "Input notes whose existence in the chain is not verified by the transaction kernel, but instead is delegated to the rollup kernels."

Reply:
changed